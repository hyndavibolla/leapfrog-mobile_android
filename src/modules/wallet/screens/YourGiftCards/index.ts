export { default as YourGiftCards } from './YourGiftCards';
