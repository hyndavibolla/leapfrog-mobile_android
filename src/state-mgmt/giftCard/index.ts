export { default as useGiftCardDetail } from './giftCardDetail';
export { default as useGiftCards } from './giftCards';
export { default as useGifCardApplePass } from './giftCardApplePass';
export { default as useGifCardAndroidPass } from './giftCardAndroidPass';
export { default as useGiftCardBalance } from './giftCardBalance';
export { default as useGiftCardStatus } from './giftCardStatus';
