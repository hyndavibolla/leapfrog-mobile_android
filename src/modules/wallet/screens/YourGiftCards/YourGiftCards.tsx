import React, { memo, useContext, useCallback, useState, useLayoutEffect, useEffect } from 'react';
import { View, FlatList, Text } from 'react-native';
import { StackNavigationProp } from '@react-navigation/stack';
import { useFocusEffect } from '@react-navigation/native';

import { Button } from '_components/Button';
import { GiftCard } from '_views/WalletMain/components/GiftCard';
import { SwipeItem } from '_commons/components/molecules/SwipeItem';
import { Icon } from '_commons/components/atoms/Icon';
import { StatusConfirmationModal } from '_modules/wallet/components/StatusConfirmationModal';
import { getPageNameByRoute } from '_utils/trackingUtils';

import { GlobalContext } from '_state_mgmt/GlobalState';
import { useTestingHelper } from '_utils/useTestingHelper';
import { useGiftCardStatus } from '_state_mgmt/giftCard';
import { IGiftCard, statusType, IGiftCardStatus } from '_models/giftCard';
import { ROUTES, ICON, ForterActionType, PageNames, TealiumEventType, UxObject, COLOR, FONT_SIZE } from '_constants';

import { TouchableOpacity } from 'react-native-gesture-handler';
import { useEventTracker } from '_state_mgmt/core/hooks';

import { styles } from './styles';

export interface Props {
  navigation: StackNavigationProp<any>;
  route?: {
    params: {
      showYourGiftCards?: boolean;
    };
  };
}

export const YourGiftCards = ({ navigation, route }: Props) => {
  const { showYourGiftCards } = route.params;
  const { trackUserEvent, trackView } = useEventTracker();
  const { getTestIdProps } = useTestingHelper('wallet-your-gc');
  const {
    state: {
      giftCard: { giftCardsList }
    },
    deps: {
      nativeHelperService: { platform },
      eventTrackerService: { rZero }
    }
  } = useContext(GlobalContext);

  const [isGiftCardsActive, setIsGiftCardsActive] = useState<boolean>(true);
  const [giftCardsFiltered, setGiftCardsFiltered] = useState<IGiftCard[]>([]);
  const [isVisibleModal, setIsVisibleModal] = useState<boolean>(false);
  const [giftCardName, setGiftCardName] = useState<string>('');
  const [giftCardStatus, setGiftCardStatus] = useState<IGiftCardStatus>({ id: '', status: undefined, cardProvider: '' });

  const [setCardStatus, isLoading] = useGiftCardStatus();

  const trackEvent = useCallback(
    (isEntering: boolean) => {
      const currentRouteName = isGiftCardsActive
        ? ROUTES.WALLET_YOUR_GIFT_CARDS.TAB_YOUR_GIFT_CARDS_ACTIVE
        : ROUTES.WALLET_YOUR_GIFT_CARDS.TAB_YOUR_GIFT_CARDS_ARCHIVED;

      if (isEntering) trackView(currentRouteName as string);
      rZero.trackView(getPageNameByRoute(currentRouteName), isEntering);
    },
    [isGiftCardsActive, rZero, trackView]
  );

  useFocusEffect(
    useCallback(() => {
      return () => trackEvent(false);
    }, [trackEvent])
  );

  useLayoutEffect(() => {
    const giftCards = giftCardsList.filter(({ statusInd }) => statusInd === (isGiftCardsActive ? statusType.ACTIVE : statusType.HIDDEN));
    setGiftCardsFiltered(giftCards);
    trackEvent(true);
  }, [giftCardsList, isGiftCardsActive, trackEvent]);

  useEffect(() => {
    if (showYourGiftCards !== undefined) setIsGiftCardsActive(showYourGiftCards);
  }, [showYourGiftCards]);

  const keyExtractor = useCallback((item: IGiftCard) => `${item.providerCardId}-${item.providerBrandId}`, []);

  const goToGiftCardDetail = useCallback(
    ({
      providerCardId,
      brandName,
      cardBalance,
      statusInd,
      cardProvider
    }: {
      providerCardId: string;
      brandName: string;
      cardBalance: number;
      statusInd: statusType;
      cardProvider: string;
    }) =>
      navigation.navigate(ROUTES.GIFT_CARD_LIST_DETAIL, {
        giftCardId: providerCardId,
        title: brandName,
        cardBalance,
        statusInd,
        cardProvider
      }),
    [navigation]
  );

  const onTapHandle = useCallback((id: string, cardProvider: string, status: statusType, name: string) => {
    setGiftCardStatus({ id, status, cardProvider });
    setGiftCardName(name);
    setIsVisibleModal(true);
  }, []);

  const onSuccessHandle = useCallback(async () => {
    const newStatus = giftCardStatus.status === statusType.ACTIVE ? statusType.HIDDEN : statusType.ACTIVE;
    await setCardStatus(giftCardStatus.id, newStatus, giftCardStatus.cardProvider, 150);
    trackUserEvent(
      TealiumEventType.WALLET,
      {
        page_name: isGiftCardsActive ? PageNames.WALLET.WALLET_YOUR_GIFT_CARDS_ACTIVE : PageNames.WALLET.WALLET_YOUR_GIFT_CARDS_ARCHIVED,
        event_name: TealiumEventType.WALLET,
        uxObject: UxObject.BUTTON,
        event_type: giftCardStatus.status === statusType.ACTIVE ? TealiumEventType.GIFT_CARD_ARCHIVE : TealiumEventType.GIFT_CARD_UNARCHIVE
      },
      ForterActionType.TAP
    );

    setIsVisibleModal(false);
  }, [giftCardStatus, setCardStatus, isGiftCardsActive, trackUserEvent]);

  const renderItem = useCallback(
    ({
      item: {
        providerCardId,
        providerBrandId,
        cardValue,
        cardBalance,
        cardBalanceCheckDt,
        purchaseTs,
        statusInd,
        cardProvider,
        brandDetails: { brandName, brandLogo, faceplateUrl }
      }
    }) => {
      const isActiveGiftCard = statusInd === statusType.ACTIVE;
      const key = `${providerCardId}-${providerBrandId}`;
      if (platform.OS === 'android') {
        return (
          <GiftCard
            key={key}
            name={brandName}
            value={cardValue}
            cardBalance={cardBalance}
            cardBalanceCheckDt={cardBalanceCheckDt}
            purchaseTs={purchaseTs}
            onPress={() => goToGiftCardDetail({ providerCardId, brandName, cardBalance, statusInd, cardProvider })}
            onLongPress={() => onTapHandle(providerCardId, cardProvider, statusInd, brandName)}
            brandLogo={brandLogo}
            faceplateUrl={faceplateUrl}
          />
        );
      }
      return (
        <SwipeItem
          key={key}
          onTapRight={() => onTapHandle(providerCardId, cardProvider, statusInd, brandName)}
          iconLeft={isActiveGiftCard ? ICON.FOLDER_ARROW_DOWN : ICON.FOLDER_ARROW_UP}
          textLeft={isActiveGiftCard ? 'Archive' : 'Unarchive'}
          backgroundColor={isActiveGiftCard ? COLOR.DARK_GRAY : COLOR.PRIMARY_BLUE}
        >
          <GiftCard
            name={brandName}
            value={cardValue}
            cardBalance={cardBalance}
            cardBalanceCheckDt={cardBalanceCheckDt}
            purchaseTs={purchaseTs}
            onPress={() => goToGiftCardDetail({ providerCardId, brandName, cardBalance, statusInd, cardProvider })}
            brandLogo={brandLogo}
            faceplateUrl={faceplateUrl}
          />
        </SwipeItem>
      );
    },
    [goToGiftCardDetail, onTapHandle, platform.OS]
  );

  return (
    <View style={styles.container} {...getTestIdProps('container')}>
      <View style={styles.toggleButtonContainer}>
        <View style={styles.toggleButton}>
          <TouchableOpacity {...getTestIdProps('see-actives')} onPress={() => setIsGiftCardsActive(true)}>
            <Text style={[styles.toggleButtonText, isGiftCardsActive && styles.toggleButtonTextActive]}>Your Gift Cards</Text>
          </TouchableOpacity>
        </View>
        <View style={styles.toggleButtonSeparator} />
        <View style={styles.toggleButton}>
          <TouchableOpacity {...getTestIdProps('see-archives')} onPress={() => setIsGiftCardsActive(false)}>
            <Text style={[styles.toggleButtonText, !isGiftCardsActive && styles.toggleButtonTextActive]}>Archived Gift Cards</Text>
          </TouchableOpacity>
        </View>
      </View>

      {!giftCardsFiltered.length ? (
        <View style={styles.fallback} {...getTestIdProps('fallback')}>
          <Icon
            name={ICON.EMPTY_STATE}
            size={FONT_SIZE.XL}
            color={isGiftCardsActive ? COLOR.PRIMARY_BLUE : COLOR.DARK_GRAY}
            backgroundStyle={[styles.iconBackground, { backgroundColor: isGiftCardsActive ? COLOR.SOFT_PRIMARY_BLUE : COLOR.MEDIUM_GRAY }]}
            innerBackgroundStyle={styles.iconInnerBackground}
          />
          <Text style={styles.titleFallback}>
            {isGiftCardsActive
              ? "Uh-oh! It looks like you don't have any active Gift Cards yet."
              : "Uh-oh! It looks like you don't have any archived Gift Cards yet."}
          </Text>
        </View>
      ) : (
        <FlatList data={giftCardsFiltered} keyExtractor={keyExtractor} renderItem={renderItem} showsVerticalScrollIndicator={false} />
      )}
      <View style={styles.footerContainer}>
        <Button
          innerContainerStyle={styles.innerButton}
          textStyle={styles.buttonText}
          onPress={() => navigation.navigate(ROUTES.MAIN_TAB.REWARDS)}
          {...getTestIdProps('buy-gift-card')}
        >
          Buy Gift Card <Icon name={ICON.REWARDS_GIFT_CARDS} size={FONT_SIZE.SMALLER} color={COLOR.WHITE} />
        </Button>
      </View>
      <StatusConfirmationModal
        name={giftCardName}
        isVisible={isVisibleModal}
        isLoading={isLoading}
        status={giftCardStatus.status}
        onDismiss={() => setIsVisibleModal(false)}
        onSuccess={onSuccessHandle}
      />
    </View>
  );
};

export default memo(YourGiftCards);
