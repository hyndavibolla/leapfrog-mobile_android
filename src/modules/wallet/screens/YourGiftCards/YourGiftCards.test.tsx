import React from 'react';
import { fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import { Deps, IGlobalState } from '_models/general';
import { getInitialState } from '_state_mgmt/GlobalState';
import { getMockDeps } from '_test_utils/getMockDeps';

import { ROUTES } from '_constants/routes';
import { getGiftCard } from '_test_utils/entities';
import { statusType } from '_models/giftCard';

import YourGiftCards, { Props } from './YourGiftCards';

describe('YourGiftCards', () => {
  let props: Props;
  let state: IGlobalState;
  let deps: Deps;
  let render;

  beforeEach(() => {
    state = getInitialState();
    deps = getMockDeps();
    props = {
      navigation: { navigate: jest.fn() } as any,
      route: {
        params: {
          showYourGiftCards: true
        }
      }
    };
    const { Screen, Navigator } = createStackNavigator();
    render = (newProps, newDeps, newState) => {
      const Component = () => <YourGiftCards {...props} {...(newProps || {})} />;
      return renderWithGlobalContext(
        <NavigationContainer>
          <Navigator>
            <Screen name="route" component={Component} />
          </Navigator>
        </NavigationContainer>,
        newDeps,
        newState
      );
    };
  });

  it('should render without param showYourGiftCards', () => {
    props.route.params.showYourGiftCards = undefined;
    deps.apiService.fetchActivityHistory = jest.fn().mockResolvedValue([getGiftCard()]);
    const { toJSON } = render(<YourGiftCards {...props} />, deps, state);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should NOT render', () => {
    state.giftCard.giftCardsList = [];
    const { queryByTestId } = render(<YourGiftCards {...props} />, deps, state);
    expect(queryByTestId('gift-card-container')).toBeNull();
  });

  it('should navigate to details', () => {
    state.giftCard.giftCardsList = [getGiftCard()];
    const { getAllByTestId } = render(<YourGiftCards {...props} />, deps, state);
    fireEvent.press(getAllByTestId('gift-card-container')[0]);
    expect(props.navigation.navigate).toBeCalledWith(
      ROUTES.GIFT_CARD_LIST_DETAIL,
      expect.objectContaining({ giftCardId: expect.any(String), title: expect.any(String) })
    );
  });

  it('should navigate to rewards tab', () => {
    const { getByTestId } = render(<YourGiftCards {...props} />, deps, state);
    fireEvent.press(getByTestId('wallet-your-gc-buy-gift-card'));
    expect(props.navigation.navigate).toBeCalledWith(ROUTES.MAIN_TAB.REWARDS);
  });

  it('should not render swipeable component', () => {
    deps.nativeHelperService.platform.OS = 'android';
    state.giftCard.giftCardsList = [getGiftCard()];
    const { queryByTestId } = render(<YourGiftCards {...props} />, deps, state);
    expect(queryByTestId('swipe-item-element')).toBeNull();
  });

  it('should action swipe (ios)', () => {
    deps.nativeHelperService.platform.OS = 'ios';
    state.giftCard.giftCardsList = [getGiftCard()];
    const { getAllByTestId, getByTestId } = render(<YourGiftCards {...props} />, deps, state);
    fireEvent.press(getAllByTestId('swipe-item-right-button')[0]);
    expect(getByTestId('archive-confirmation-modal-content')).toBeTruthy();
  });

  it('should not action swipe show up on android', () => {
    deps.nativeHelperService.platform.OS = 'android';
    state.giftCard.giftCardsList = [getGiftCard()];
    const { queryAllByTestId } = render(<YourGiftCards {...props} />, deps, state);
    expect(queryAllByTestId('swipe-item-right-button')[0]).toBeUndefined();
  });

  it('should navigate to details (android)', () => {
    deps.nativeHelperService.platform.OS = 'android';
    state.giftCard.giftCardsList = [getGiftCard()];
    const { getAllByTestId } = render(<YourGiftCards {...props} />, deps, state);
    fireEvent.press(getAllByTestId('gift-card-container')[0]);
    expect(props.navigation.navigate).toBeCalledWith(
      ROUTES.GIFT_CARD_LIST_DETAIL,
      expect.objectContaining({ giftCardId: expect.any(String), title: expect.any(String) })
    );
  });

  it('should action a long press (android)', () => {
    deps.nativeHelperService.platform.OS = 'android';
    state.giftCard.giftCardsList = [getGiftCard()];
    const { getAllByTestId, getByTestId } = render(<YourGiftCards {...props} />, deps, state);
    fireEvent(getAllByTestId('gift-card-container')[0], 'longPress');
    expect(getByTestId('archive-confirmation-modal-content')).toBeTruthy();
  });

  it('should not action a long press show up on ios', () => {
    deps.nativeHelperService.platform.OS = 'ios';
    state.giftCard.giftCardsList = [getGiftCard()];
    const { queryAllByTestId } = render(<YourGiftCards {...props} />, deps, state);
    expect(queryAllByTestId('gift-card-container')[0]).not.toHaveProp('longPress');
  });

  it('should render the fallback when there is not gift cards actives', () => {
    state.giftCard.giftCardsList = [];
    const { getByTestId } = render(<YourGiftCards {...props} />, deps, state);
    fireEvent(getByTestId('wallet-your-gc-see-actives'), 'onPress');
    expect(getByTestId('wallet-your-gc-fallback')).toBeTruthy();
  });

  it('should render the fallback when there is not gift cards archive', () => {
    state.giftCard.giftCardsList = [];
    const { getByTestId } = render(<YourGiftCards {...props} />, deps, state);
    fireEvent(getByTestId('wallet-your-gc-see-archives'), 'onPress');
    expect(getByTestId('wallet-your-gc-fallback')).toBeTruthy();
  });

  it('should not render the fallback when there is gift cards archive', () => {
    state.giftCard.giftCardsList = [{ ...getGiftCard(), statusInd: statusType.HIDDEN }];
    const { queryByTestId, getByTestId } = render(<YourGiftCards {...props} />, deps, state);
    fireEvent(getByTestId('wallet-your-gc-see-archives'), 'onPress');
    expect(queryByTestId('wallet-your-gc-fallback')).toBeNull();
  });

  it('should dismiss modal with cancel button', () => {
    deps.nativeHelperService.platform.OS = 'ios';
    state.giftCard.giftCardsList = [getGiftCard()];

    const { getAllByTestId, getByTestId, queryByTestId } = render(<YourGiftCards {...props} />, deps, state);
    fireEvent.press(getAllByTestId('swipe-item-right-button')[0]);
    expect(getByTestId('archive-confirmation-modal-content')).toBeTruthy();
    fireEvent.press(getByTestId('archive-confirmation-modal-cancel-button'));
    expect(queryByTestId('archive-confirmation-modal')).toBeNull();
  });

  it('should dismiss modal when press confirm button and giftcard status is active', async () => {
    deps.nativeHelperService.platform.OS = 'ios';
    state.giftCard.giftCardsList = [getGiftCard()];

    const { getAllByTestId, getByTestId, queryByTestId } = render(<YourGiftCards {...props} />, deps, state);
    fireEvent.press(getAllByTestId('swipe-item-right-button')[0]);
    expect(getByTestId('archive-confirmation-modal-content')).toBeTruthy();
    fireEvent.press(getByTestId('archive-confirmation-modal-confirm-button'));
    expect(deps.apiService.updateCardStatus).toBeCalledWith({
      id: getGiftCard().providerCardId,
      status: statusType.HIDDEN,
      cardProvider: getGiftCard().cardProvider
    });
    await waitFor(() => {
      expect(queryByTestId('archive-confirmation-modal')).toBeNull();
    });
  });

  it('should dismiss modal when press confirm button and giftcard status is hidden', async () => {
    deps.nativeHelperService.platform.OS = 'ios';
    state.giftCard.giftCardsList = [{ ...getGiftCard(), statusInd: statusType.HIDDEN }];

    const { getAllByTestId, getByTestId, queryByTestId } = render(<YourGiftCards {...props} />, deps, state);
    fireEvent.press(getByTestId('wallet-your-gc-see-archives'));
    fireEvent.press(getAllByTestId('swipe-item-right-button')[0]);
    expect(getByTestId('archive-confirmation-modal-content')).toBeTruthy();
    fireEvent.press(getByTestId('archive-confirmation-modal-confirm-button'));
    expect(deps.apiService.updateCardStatus).toBeCalledWith({
      id: getGiftCard().providerCardId,
      status: statusType.ACTIVE,
      cardProvider: getGiftCard().cardProvider
    });
    await waitFor(() => expect(queryByTestId('archive-confirmation-modal')).toBeNull());
  });
});
