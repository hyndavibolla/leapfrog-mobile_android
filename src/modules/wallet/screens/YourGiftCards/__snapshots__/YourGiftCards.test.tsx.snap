// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourGiftCards should render without param showYourGiftCards 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          collapsable={false}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          collapsable={false}
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                collapsable={false}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                route
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        collapsable={false}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            collapsable={false}
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      accessibilityLabel="wallet-your-gc-container"
                      accessible={true}
                      style={
                        Object {
                          "backgroundColor": "#F5F7FA",
                          "flexBasis": 0,
                          "flexGrow": 1,
                          "flexShrink": 1,
                        }
                      }
                      testID="wallet-your-gc-container"
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#FFFFFF",
                            "borderBottomLeftRadius": 52,
                            "borderBottomRightRadius": 52,
                            "borderTopLeftRadius": 52,
                            "borderTopRightRadius": 52,
                            "elevation": 4,
                            "flexDirection": "row",
                            "height": 52,
                            "marginBottom": 26,
                            "marginLeft": 16,
                            "marginRight": 16,
                            "marginTop": 26,
                            "shadowColor": "#262C2F",
                            "shadowOffset": Object {
                              "height": 2,
                              "width": 0,
                            },
                            "shadowOpacity": 0.2,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "flexShrink": 1,
                            }
                          }
                        >
                          <RNGestureHandlerButton
                            collapsable={false}
                            onGestureEvent={[Function]}
                            onGestureHandlerEvent={[Function]}
                            onGestureHandlerStateChange={[Function]}
                            onHandlerStateChange={[Function]}
                            rippleColor={0}
                            testID="wallet-your-gc-see-actives"
                          >
                            <View
                              accessibilityLabel="wallet-your-gc-see-actives"
                              accessible={true}
                              collapsable={false}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#909BA0",
                                      "fontFamily": "SFProDisplay-SemiBold",
                                      "fontSize": 14,
                                    },
                                    Object {
                                      "color": "#0066CC",
                                    },
                                  ]
                                }
                              >
                                Your Gift Cards
                              </Text>
                            </View>
                          </RNGestureHandlerButton>
                        </View>
                        <View
                          style={
                            Object {
                              "borderRightColor": "#E2E3E3",
                              "borderRightWidth": 1,
                              "height": 40,
                            }
                          }
                        />
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "flexShrink": 1,
                            }
                          }
                        >
                          <RNGestureHandlerButton
                            collapsable={false}
                            onGestureEvent={[Function]}
                            onGestureHandlerEvent={[Function]}
                            onGestureHandlerStateChange={[Function]}
                            onHandlerStateChange={[Function]}
                            rippleColor={0}
                            testID="wallet-your-gc-see-archives"
                          >
                            <View
                              accessibilityLabel="wallet-your-gc-see-archives"
                              accessible={true}
                              collapsable={false}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#909BA0",
                                      "fontFamily": "SFProDisplay-SemiBold",
                                      "fontSize": 14,
                                    },
                                    false,
                                  ]
                                }
                              >
                                Archived Gift Cards
                              </Text>
                            </View>
                          </RNGestureHandlerButton>
                        </View>
                      </View>
                      <View
                        accessibilityLabel="wallet-your-gc-fallback"
                        accessible={true}
                        style={
                          Object {
                            "alignItems": "center",
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "flexShrink": 1,
                            "marginTop": 80,
                          }
                        }
                        testID="wallet-your-gc-fallback"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "borderRadius": 60,
                                "height": 105,
                                "justifyContent": "center",
                                "width": 105,
                              },
                              Array [
                                Object {
                                  "alignItems": "flex-end",
                                  "backgroundColor": "#DDE8F5",
                                },
                                Object {
                                  "backgroundColor": "#DDE8F5",
                                },
                              ],
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "backgroundColor": "#FFFFFF",
                                "borderBottomLeftRadius": 70,
                                "borderBottomRightRadius": 70,
                                "borderTopLeftRadius": 70,
                                "borderTopRightRadius": 70,
                              }
                            }
                          >
                            <Text
                              accessibilityLabel="icon-text"
                              accessible={true}
                              allowFontScaling={false}
                              selectable={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#0066CC",
                                    "fontSize": 60,
                                  },
                                  undefined,
                                  Object {
                                    "fontFamily": "sywmax",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                              testID="icon-text"
                            >
                              
                            </Text>
                          </View>
                        </View>
                        <Text
                          style={
                            Object {
                              "color": "#262C2F",
                              "fontFamily": "SFProDisplay-Bold",
                              "fontSize": 16,
                              "lineHeight": 20,
                              "marginBottom": 0,
                              "marginLeft": 36,
                              "marginRight": 36,
                              "marginTop": 32,
                              "textAlign": "center",
                            }
                          }
                        >
                          Uh-oh! It looks like you don't have any active Gift Cards yet.
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "height": 100,
                            "justifyContent": "center",
                            "paddingHorizontal": 32,
                          }
                        }
                      >
                        <View
                          accessibilityLabel="wallet-your-gc-buy-gift-card"
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID="wallet-your-gc-buy-gift-card"
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#0066CC",
                                  "borderBottomLeftRadius": 30,
                                  "borderBottomRightRadius": 30,
                                  "borderTopLeftRadius": 30,
                                  "borderTopRightRadius": 30,
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                  "paddingBottom": 8,
                                  "paddingLeft": 12,
                                  "paddingRight": 12,
                                  "paddingTop": 8,
                                },
                                Object {
                                  "paddingBottom": 16,
                                  "paddingLeft": 0,
                                  "paddingRight": 0,
                                  "paddingTop": 16,
                                },
                              ]
                            }
                          >
                            <View
                              style={false}
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#909BA0",
                                      "fontFamily": "SFProDisplay-Medium",
                                      "fontSize": 16,
                                      "lineHeight": 22,
                                    },
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontSize": 14,
                                        "textAlign": "center",
                                      },
                                      Object {
                                        "fontFamily": "SFProDisplay-SemiBold",
                                        "fontSize": 14,
                                        "lineHeight": 20,
                                      },
                                      undefined,
                                    ],
                                  ]
                                }
                              >
                                Buy Gift Card 
                              </Text>
                            </View>
                            <View
                              style={
                                Object {
                                  "marginLeft": 8,
                                }
                              }
                            >
                              <Text
                                accessibilityLabel="icon-text"
                                accessible={true}
                                allowFontScaling={false}
                                selectable={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontSize": 16,
                                    },
                                    undefined,
                                    Object {
                                      "fontFamily": "sywmax",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                                testID="icon-text"
                              >
                                
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                      <Modal
                        accessibilityLabel="modal-container"
                        accessible={true}
                        animationType="slide"
                        hardwareAccelerated={false}
                        statusBarTranslucent={true}
                        testID="modal-container"
                        transparent={true}
                        visible={false}
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </RNCSafeAreaProvider>
</View>
`;
